### DISCUSSION TITLE
closer and closer to publication

### DISCUSSION THREAD

**Discussion by philipschwarz** [winitzki/sofp#88] [General]
Hello Sergei - just a line to say that the frequent github notifications that you are working on completing the book are for me a pleasant/exciting reminder that the book is getting closer and closer to publication. I look forward to that day!

**Comment by winitzki** [comment]
Hi Philip,

Thank you for writing. The publication will be unfortunately delayed. I need to reformat the book into 3 volumes because the size of the book has grown to be too large for a single volume. Also, the book currently has a very small print and complicated typography; it's not ideal. I am reformatting the book in 12 point font, and this means a complete editing pass over the entire text. Right now this pass is at around chapter 6.

Finally, there remains some unfinished work in chapter 13, although I believe I finally resolved all outstanding issues and I have full proofs of what I want to prove. This year I will make an effort to finish things up.

**Reply by philipschwarz** [reply]
Thank you for the update. 

Looking forward to the volumes, whenever they come out.

As is said in Chapter 2 of The Mythical Man-Month, 

> Good cooking takes time. If you are made to wait, it is to serve you better, and to please you.—MENU OF RESTAURANT ANTOINE, NEW ORLEANS

![image](https://github.com/winitzki/sofp/assets/116804/f37f856b-054a-4ecc-85a1-a6678c05141c)

**Reply by winitzki** [reply]
Since my daughter was born, the work on the book has slowed to a crawl. But the work is progressing and there is no obstacle at the moment to finishing.

The reformatting for 12pt font is going to be easier to manage because I'm abandoning most of the boxed-code or boxed-diagrams layout.

I will first publish parts 1 and 2 as volumes 1 and 2 and then continue working on part 3, which in any case very few people will want to read.

**Comment by akiemw** [comment]
<img width="894" alt="Screenshot 2025-04-01 at 11 32 55 AM" src="https://github.com/user-attachments/assets/65818ea5-c7c6-421d-b9f2-533d6144ad8b" />
I believe this question could use some clarification, something around the "such that there are at least three different integers j" How is it that the set (3, 4, 5) is not three distinct j such that 1 <= j <= k and set(3, 4, 5).size >= 3 therefore 5 should be included in the example solution as mines just test for all that satisfy the two conditions 1 <= j <= k and j * j > 2 * k.

**Reply by winitzki** [reply]
>   I believe this question could use some clarification, something around the "such that there are at least three different integers j" How is it that the set (3, 4, 5) is not three distinct j such that 1 <= j <= k and set(3, 4, 5).size >= 3 therefore 5 should be included in the example solution as mines just test for all that satisfy the two conditions 1 <= j <= k and j * j > 2 * k.

@akiemw  In your view, what is not clear in the question?

Your solution finds all `k` such that  at least one `j` exists such that `1 <= j <= k` and `j * j > 2 * k`.

The question asks for `k` such that at least three `j` exist such that `1 <= j <= k` and `j * j > 2 * k`.
It is not the same condition.

Consider k = 5. What are all `j` such that `1 <= j <= 5` and `j * j > 2 * 5`? Those are `j = 4` and `j = 5` only. The value `j = 3` does not satisfy `j * j > 2 * 5`.

**Reply by akiemw** [reply]
I see the flaw in my thinking now, and thanks for the quick response because I was really racking my brain behind what I was failing to see there, but just required more careful attention to details. Well they say if math was easy everyone would be doing it. So, I'll try harder. Think what I kept missing was that I have to hold `k` constant as I loop through my `j`'s. Which I saw you doing in your example and I guess I just jumped the gun because I received an output in my attempt to solve the example without reviewing the solution and expected to receive an error if my attempt was incorrect. But as I know you can get the arithmetic perfect and still arrive at the wrong solution. Case and point.
